### ISSUE TITLE
Core Game Idea

### ISSUE THREAD

**Issue by Den4200** [Den4200/game-jam-2020#19]
## Description
We need a core game idea. We've come up with many different ideas, but haven't really discussed or formally agreed on any of them. However currently, we have an incredibly good base game.

## Current Game Implementation
The player has 3 different wizards to pick from: red, green, or blue. Each colored wizard has their own special abilities.

| Color | Description | Attack Multiplier ( /\ ) | Speed Multiplier ( /\ ) | Resistance ( /\ ) | Dexterity ( \\/ ) |
| --- | --- | --- | --- | --- | --- |
| Red   | Strong and powerful, destroying all who dare defy him. | 1.5 | 1.1 | 0.1 | 0.6 |
| Green | Built like a tank and as strong as a brink, this wizard can take a punch. | 1.1 | 1 | 0.5 | 0.75 |
| Blue | Quick and fast, speeding through enemy ranks. | 1 | 1.5 | 0 | 0.5 |

*Note: ( /\ ) means a higher value is better and ( \\/ ) means a lower value is better*

When hovering over a card to select a specific colored wizard, the game slows down time, making everything move in slow motion.

The player can move around the map by clicking on a tile. The wizard will then use A* pathfinding to figure out its way over to the player's selected destination.

The map is generated using cellular automata through several generations. The map consists of three things, The flooring, walls, and spikes. The spikes periodically extend up out of the ground, and if the player is on top of them, they will be damaged.

The player and some specific enemies are able to shoot laser projectiles at each other, causing damage to the one entity hit. Currently, we have two enemies: The chasing demons and stationary imps. The chasing demons will chase the player if within range, and use melee attacks. On the other hand, stationary imps will *not* chase the player. Instead, they will sit still and fire lasers when the player is within range.

I think that's about it for our current game implementation.

## Figure Out the Core Design & End Goal
What will our core design and end goal be for this game? Will the game's idea be centered around "survive as long as you can"? Or will it be more like a strategic puzzle game, solving levels and fighting enemies along the way? This game could branch off into so many directions, as it is in a very flexible stage. We need to nail down our idea and final goal.

## Some Ideas
- Have the game be "survive as long as you can" with multiple waves of enemies spawning in. There could possibly be power-up items to pick up and use, boosting the player's abilities in some way. In the end, when the player dies, the score will be uploaded to a server to be added into a leader board.
    - Extending on this idea, we can have multiplayer games where 2 or more players can work together to survive the longest. When the players both die, the score will be saved into a different leader board than singleplayer.
   - Extending on this idea even further, we can have multiplayer games where 2 or more players can battle each other, in a free-for-all.

## Conclusion
We need ideas and we need to agree on one! Discuss below.

**Comment by F4zii** [comment]
I like the puzzle idea much more, but we can combine these genres, so the player will have to discover and travel in the map, switch between the characters to survive and finish levels.

as you guys already saw, I really like the idea of having vision for each player(according to the game name), so maybe we can add that to (one of the main mechanic of the puzzle game):
![image](https://user-images.githubusercontent.com/44242259/79955227-284fef00-8487-11ea-8d9d-38f65f9fe1d3.png)
So I would love if we consider adding that into the game.

**Some more notes about this idea:**
All items and entities are collidable, items not collectable if not in the same color

quick exmaple: You need to collect 3 keys/orbs to pass each room, you have enemies, secret doors and some attractions (conversations and more stuff we will add)
![image](https://user-images.githubusercontent.com/44242259/79955492-8d0b4980-8487-11ea-8449-219a15e5dca6.png)

**Comment by Den4200** [comment]
The puzzle idea would definitely be interesting, but I'm not sure if that's feasible. We'd have to rip out our whole current map implementation and manually create all those levels. We wouldn't need to just create one level, but *at the very least* 5 or more. We could decide to make a procedural dungeon map generator, but that would still take a *lot* of work to make it dynamic. On top of that, we'd need to also implement all of those features you came up with, which I don't think we'd have enough time to do. 

However, at our current state, I think we can implement *some* of those features in, such as different vision for each wizard. The red wizard could possibly see power ups to boost abilities, the green wizard could see traps to avoid them, and the blue wizard could see something else.

**Comment by F4zii** [comment]
Yeah, i understand that, so we will add some features from here and there and stick to the current idea

**Comment by albertopoljak** [comment]
As discussed previously, I think the most feasible approach would be some kind of survial thing, where we can define for each level how many enemies will there be and what types.

For example first level would be 10 enemies and 2 types, next 20 enemies and 3 types etc
We can have boss level each X levels (3?) with generated map that has bigger spaces and 1 uniqe strong enemy.

For the vision thing we can partialy implement previous ideas, for example certain player can see traps, or maybe all can see them but in short radius(only when they are really close), and that one can see in bigger radius from far away.

The puzzle thing is really hard as levels would need to be designed and we didn't ever have a concrete example/plan for that and at this time duration there is no time. 
The level design for that would simply take too much time, it's like Portal - we have the idea, the portal gun, but no actual idea on how the levels will look (and level design would be 99% of gameplay in the case of logical game)
